.github/workflows/iroha2-release.yml

```diff
@@ -1,40 +1,52 @@
 name: I2::Release::Publish
 
 on:
-  push:
-    branches:
-      - iroha2-stable
-      - iroha2-lts
+  workflow_run:
+    workflows: [I2::Base::Publish]
+    types: [completed]
+    branches: [iroha2-stable, iroha2-lts]
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  dockerhub:
-    runs-on: [self-hosted, Linux]
-    container:
-      image: 7272721/i2-ci:nightly
+  tag:
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         tag: [stable, lts]
-    if: github.ref == 'refs/heads/iroha2-${{ matrix.tag }}'
+    outputs:
+      tag: ${{ steps.tag.outputs.tag }}
+    steps:
+      - id: tag
+        if: github.ref == 'refs/heads/iroha2-${{ matrix.tag }}'
+        run: echo "::set-output name=tag::${{ matrix.tag }}"
+
+  dockerhub:
+    needs: tag
+    runs-on: ubuntu-latest
+    container:
+      image: 7272721/i2-ci:nightly
```

Comment:
We're getting a little overreliant on `matrix`, which contradicts both the simplicity and the performance gains from using a custom image.

Reply:
I agree. It would run as many jobs as the number of combinations in the matrix.
Previously I assumed that when used in `<job_id>.if` it could filter out unrelated jobs, but it turned out the matrix values were not evaluated at that point.
I'll take another way